### PR TITLE
Move legacy free pref to AppPreferences

### PR DESCRIPTION
**Pull request by ezaquarii**
Finally, the beast is dead...

Signed-off-by: Chris Narkiewicz <hello@ezaquarii.com>

### CODE REVIEW

**Comment by AndyScherzinger** [comment]
Hi @ezaquarii, it doesn't compile in all build flavors:

```
/drone/src/github.com/nextcloud/android/src/main/java/com/owncloud/android/ui/activity/SyncedFoldersActivity.java:190: error: constructor SyncedFolderProvider in class SyncedFolderProvider cannot be applied to given types;
        mSyncedFolderProvider = new SyncedFolderProvider(getContentResolver());
                                ^
  required: ContentResolver,AppPreferences
  found: ContentResolver
  reason: actual and formal argument lists differ in length
/drone/src/github.com/nextcloud/android/src/main/java/com/owncloud/android/jobs/MediaFoldersDetectionJob.java:83: error: constructor SyncedFolderProvider in class SyncedFolderProvider cannot be applied to given types;
        SyncedFolderProvider syncedFolderProvider = new SyncedFolderProvider(contentResolver);
                                                    ^
  required: ContentResolver,AppPreferences
  found: ContentResolver
  reason: actual and formal argument lists differ in length
/drone/src/github.com/nextcloud/android/src/main/java/com/owncloud/android/jobs/AccountRemovalJob.java:95: error: constructor SyncedFolderProvider in class SyncedFolderProvider cannot be applied to given types;
            SyncedFolderProvider syncedFolderProvider = new SyncedFolderProvider(context.getContentResolver());
                                                        ^
  required: ContentResolver,AppPreferences
  found: ContentResolver
  reason: actual and formal argument lists differ in length
/drone/src/github.com/nextcloud/android/src/main/java/com/owncloud/android/jobs/FilesSyncJob.java:106: error: constructor SyncedFolderProvider in class SyncedFolderProvider cannot be applied to given types;
        SyncedFolderProvider syncedFolderProvider = new SyncedFolderProvider(contentResolver);
                                                    ^
  required: ContentResolver,AppPreferences
  found: ContentResolver
  reason: actual and formal argument lists differ in length
/drone/src/github.com/nextcloud/android/src/main/java/com/owncloud/android/jobs/NContentObserverJob.java:68: error: constructor SyncedFolderProvider in class SyncedFolderProvider cannot be applied to given types;
                new SyncedFolderProvider(getContentResolver()).countEnabledSyncedFolders() > 0) {
                ^
  required: ContentResolver,AppPreferences
  found: ContentResolver
  reason: actual and formal argument lists differ in length
/drone/src/github.com/nextcloud/android/src/main/java/com/owncloud/android/utils/FilesSyncHelper.java:160: error: constructor SyncedFolderProvider in class SyncedFolderProvider cannot be applied to given types;
        SyncedFolderProvider syncedFolderProvider = new SyncedFolderProvider(contentResolver);
                                                    ^
  required: ContentResolver,AppPreferences
  found: ContentResolver
  reason: actual and formal argument lists differ in length
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
6 errors
```

**Comment by AndyScherzinger** [comment]
I pushed a fix, let'S see if I got all missing changes :)

@ezaquarii did you run a compile before your commit/push? 😉

File: src/main/java/com/owncloud/android/jobs/AccountRemovalJob.java [hunk 1]
```diff
@@ -29,6 +29,8 @@
 
 import com.evernote.android.job.Job;
 import com.evernote.android.job.util.support.PersistableBundleCompat;
+import com.nextcloud.client.preferences.AppPreferences;
```
[the remark this answers is not in the document]
Comment (AndyScherzinger) [reply src/main/java/com/owncloud/android/jobs/AccountRemovalJob.java]: fixed

**Comment by AndyScherzinger** [comment]
Also fixed the newly introduced lint issue now. Let's see what CI has to say :)

**Comment by ezaquarii** [comment]
Embarrassing. I thought I did but it seems that I just clicked the same build target in IDE. Working late is harmful.

Let's leave it for now - I'll sort it out when I have time this evening.

**Comment by AndyScherzinger** [comment]
@ezaquarii no worries, happens to any of us! I think the PR is good to go now, would you agree?

**Comment by ezaquarii** [comment]
> @ezaquarii no worries, happens to any of us! I think the PR is good to go now, would you agree?

Looks good to me. There is 1 codacy issue but it's not mine. :)

**Comment by tobiasKaminsky** [comment]
> Finally, the beast is dead...

:tada: :tada: :tada: :tada: :tada: :tada: :tada: 

Thanks for your amazing work!

**Comment by tobiasKaminsky** [comment]
> Looks good to me. There is 1 codacy issue but it's not mine. :)

Totally fine to ignore this for now.

**Pull request merged**